src/logic/community/communities.ts

```diff
@@ -127,11 +128,39 @@ export async function createCommunityComponent(
         })
       )
 
-      const friendsAddresses = Array.from(new Set(communities.flatMap((community) => community.friends)))
+      // Filter by active voice chat if requested
+      let filteredCommunities = communitiesWithThumbnailsAndOwnerNames
+      if (options.onlyWithActiveVoiceChat) {
+        const voiceChatStatuses = await Promise.all(
+          communitiesWithThumbnailsAndOwnerNames.map(async (community) => {
+            try {
+              const status = await commsGatekeeper.getCommunityVoiceChatStatus(community.id)
+              return { communityId: community.id, isActive: status?.isActive ?? false }
+            } catch (error) {
+              // If we can't get the status, assume it's not active
+              logger.warn(`Could not get voice chat status for community ${community.id}`, {
+                error: isErrorWithMessage(error) ? error.message : 'Unknown error'
+              })
+              return { communityId: community.id, isActive: false }
+            }
+          })
+        )
```

Comment:
WDYT about moving the get voice chat status for multiple communities inside the comms gatekeeper component?